Refactoring forward_model broke integration tests

This bug was introduced in commit `9809fef` (Move forward_model out of model_config) which was part of the PR: https://github.com/equinor/ert/pull/4195

```
FAILED tests/test_ert_hooks.py::test_ecl2csv_through_ert - AssertionError: assert False
 +  where False = <bound method Path.is_file of PosixPath('OK')>()
 +    where <bound method Path.is_file of PosixPath('OK')> = PosixPath('OK').is_file
 +      where PosixPath('OK') = Path('OK')
```

--- a/src/ert/_c_wrappers/enkf/res_config.py
+++ b/src/ert/_c_wrappers/enkf/res_config.py
@@ -98,18 +98,27 @@ def parse_signature_job(signature: str) -> Tuple[str, Optional[str]]:
 
     >>> parse_signature_job("job")
     ("job", None)
+
+
+    For backwards compatability, text after first closing parenthesis is closed,
+    but a warning is displayed.
     """
 
     open_paren = signature.find("(")
     if open_paren == -1:
         return signature, None
     job_name = signature[:open_paren]
-    close_paren = signature[-1]
-    if close_paren != ")":
+    close_paren = signature.find(")")
+    if close_paren == -1:
         raise ConfigValidationError(
             f"Missing ) in FORWARD_MODEL job description {signature}"
         )
-    return job_name, signature[open_paren + 1 : -1]
+    if close_paren < len(signature):
+        logger.warning(
+            f'Arguments after closing ) in "{signature}"'
+            f' ("{signature[close_paren:]}") is ignored.'
+        )
+    return job_name, signature[open_paren + 1 : close_paren]
 
 
 class ResConfig:

--- a/src/ert/services/_base_service.py
+++ b/src/ert/services/_base_service.py
@@ -308,7 +308,7 @@ class BaseService:
             pass
         return cls.start_server(*args, **kwargs)
 
-    def wait_until_ready(self, timeout: int = None) -> bool:
+    def wait_until_ready(self, timeout: Optional[int] = None) -> bool:
         if timeout is None:
             timeout = self._timeout
 

--- a/src/ert/shared/models/multiple_data_assimilation.py
+++ b/src/ert/shared/models/multiple_data_assimilation.py
@@ -255,7 +255,7 @@ class MultipleDataAssimilation(BaseRunModel):
         self,
         run_context: RunContext,
         evaluator_server_config: EvaluatorServerConfig,
-        update_id: str = None,
+        update_id: Optional[str] = None,
     ) -> Tuple[int, str]:
         iteration = run_context.iteration
 

--- a/tests/unit_tests/c_wrappers/res/enkf/test_res_config.py
+++ b/tests/unit_tests/c_wrappers/res/enkf/test_res_config.py
@@ -19,7 +19,11 @@ from ert._c_wrappers.enkf import (
     SiteConfig,
 )
 from ert._c_wrappers.enkf.enums import HookRuntime
-from ert._c_wrappers.enkf.res_config import SINGLE_VALUE_KEYS, site_config_location
+from ert._c_wrappers.enkf.res_config import (
+    SINGLE_VALUE_KEYS,
+    parse_signature_job,
+    site_config_location,
+)
 from ert._c_wrappers.job_queue import QueueDriverEnum
 from ert._c_wrappers.sched import HistorySourceEnum
 from ert._clib.config_keywords import init_user_config_parser
@@ -850,3 +854,22 @@ def test_config_content_as_dict_single_value_keys(tmpdir):
         content_as_dict = ResConfig._config_content_as_dict(content)
         for _, value in content_as_dict.items():
             assert not isinstance(value, list)
+
+
+def test_that_parse_job_signature_passes_through_job_names():
+    assert parse_signature_job("JOB") == ("JOB", None)
+
+
+def test_that_parse_job_signature_correctly_gets_arguments():
+    assert parse_signature_job("JOB(<ARG1>=val1, <ARG2>=val2)") == (
+        "JOB",
+        "<ARG1>=val1, <ARG2>=val2",
+    )
+
+
+def test_that_parse_job_signature_warns_for_extra_parens(caplog):
+    assert parse_signature_job("JOB(<ARG1>=val1, <ARG2>=val2), <ARG3>=val3)") == (
+        "JOB",
+        "<ARG1>=val1, <ARG2>=val2",
+    )
+    assert "Arguments after closing )" in caplog.text

